src/main/java/ir/sahab/elasticsearchrule/ElasticsearchRule.java

```diff
@@ -1,141 +1,39 @@
 package ir.sahab.elasticsearchrule;
 
-import ir.sahab.cleanup.Cleanups;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.concurrent.ExecutionException;
-import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
-import org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequest;
-import org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateResponse;
-import org.elasticsearch.cli.Terminal;
 import org.elasticsearch.client.transport.TransportClient;
-import org.elasticsearch.cluster.ClusterName;
-import org.elasticsearch.cluster.health.ClusterHealthStatus;
-import org.elasticsearch.common.network.NetworkModule;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.transport.TransportAddress;
-import org.elasticsearch.env.Environment;
-import org.elasticsearch.index.reindex.ReindexPlugin;
-import org.elasticsearch.node.InternalSettingsPreparer;
-import org.elasticsearch.node.Node;
-import org.elasticsearch.node.NodeValidationException;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.transport.Netty4Plugin;
-import org.elasticsearch.transport.TransportService;
-import org.elasticsearch.transport.client.PreBuiltTransportClient;
 import org.junit.rules.ExternalResource;
 
 /**
- * A JUnit rule for starting an elasticsearch server on the local machine.
+ * A JUnit 4 {@link org.junit.rules.TestRule} for starting an elasticsearch server on the local machine.
  */
 public class ElasticsearchRule extends ExternalResource {
```

Comment:
public class ElasticsearchRule extends ElasticsearchBase implements TestRule
This way we can remove base variable and overrided methods like getAddress / getTransportClient/ waitForGreenStatus

Reply:
No, we can't. Because we use ExternalResources before and after methods.